### PR TITLE
feat(jdbc): add jdbc input support

### PR DESCRIPTION
**Pull request by lyogev**

### CODE REVIEW

File: src/main/scala/com/yotpo/metorikku/configuration/InputCollection.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+case class UnknownInputTypeException(private val message: String = "",
+                                     private val cause: Throwable = None.orNull) extends Exception(message, cause) {}
+
+case class Empty() extends Input {
+  override def getReader(name: String): ReadableInput = throw new UnknownInputTypeException()
+}
+
+trait Input {
```
Comment (RonBarabash) [inline src/main/scala/com/yotpo/metorikku/configuration/InputCollection.scala]: @lyogev please seperate traits to different file

File: src/main/scala/com/yotpo/metorikku/configuration/InputCollection.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  def getReader(name: String): ReadableInput
```
Comment (RonBarabash) [inline src/main/scala/com/yotpo/metorikku/configuration/InputCollection.scala]: @lyogev im having trouble with the names, why is it called readable input? why getReader return readable input and not reader?

File: src/main/scala/com/yotpo/metorikku/input/ReadableInput.scala [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.yotpo.metorikku.input
+
+import org.apache.spark.sql.DataFrame
+
+trait ReadableInput {
+  var name: String
```
Comment (RonBarabash) [inline src/main/scala/com/yotpo/metorikku/input/ReadableInput.scala]: @lyogev can this be val?

File: src/main/scala/com/yotpo/metorikku/input/jdbc/JDBCInput.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.yotpo.metorikku.input.ReadableInput
+import com.yotpo.metorikku.session.Session.getSparkSession
+import org.apache.spark.sql.DataFrame
+
+case class JDBCInput(var name: String, connectionUrl: String, user: String,
+                     password: String, driver: Option[String], table: String,
+                     partitionColumn: Option[String], numberOfPartitions: Option[Int]) extends ReadableInput {
+  def read(): DataFrame = {
+    val url = connectionUrl + "?zeroDateTimeBehavior=convertToNull"
```
Comment (RonBarabash) [inline src/main/scala/com/yotpo/metorikku/input/jdbc/JDBCInput.scala]: whats this?

Comment (lyogev) [reply src/main/scala/com/yotpo/metorikku/input/jdbc/JDBCInput.scala]: It's something that's only related to mysql we can move it to the client's side

Comment (lyogev) [reply src/main/scala/com/yotpo/metorikku/input/jdbc/JDBCInput.scala]: https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-reference-configuration-properties.html

**Pull request merged**